notify.go

```diff
@@ -34,9 +39,22 @@ func (p *PubSubNotif) ListenClose(n network.Network, _ ma.Multiaddr) {
 }
 
 func (p *PubSubNotif) Initialize() {
-	for _, pr := range p.host.Network().Peers() {
+	isTransient := func(pid peer.ID) bool {
+		for _, c := range p.host.Network().ConnsToPeer(pid) {
+			if !c.Stat().Transient {
+				return false
+			}
+		}
+
+		return true
+	}
```

Comment:
Gah. We really need a better peerstore/database system.

Reply:
yes, indeed!